Dalamud/Interface/ImGuiNotification/NotificationUtilities.cs

```diff
@@ -0,0 +1,154 @@
+
+        return true;
+    }
+
+    /// <summary>Draws an icon from an instance of <see cref="IDalamudTextureWrap"/>.</summary>
+    /// <param name="minCoord">The coordinates of the top left of the icon area.</param>
+    /// <param name="maxCoord">The coordinates of the bottom right of the icon area.</param>
+    /// <param name="texture">The texture.</param>
+    /// <returns><c>true</c> if anything has been drawn.</returns>
+    internal static bool DrawIconFrom(Vector2 minCoord, Vector2 maxCoord, IDalamudTextureWrap? texture)
+    {
+        if (texture is null)
+            return false;
+        try
+        {
+            var handle = texture.ImGuiHandle;
+            var size = texture.Size;
+            if (size.X > maxCoord.X - minCoord.X)
+                size *= (maxCoord.X - minCoord.X) / size.X;
+            if (size.Y > maxCoord.Y - minCoord.Y)
+                size *= (maxCoord.Y - minCoord.Y) / size.Y;
+            ImGui.SetCursorPos(((minCoord + maxCoord) - size) / 2);
+            ImGui.Image(handle, size);
+            return true;
+        }
+        catch
+        {
+            return false;
+        }
+    }
+
+    /// <summary>Draws an icon from an instance of <see cref="Task{TResult}"/> that results in an
+    /// <see cref="IDalamudTextureWrap"/>.</summary>
+    /// <param name="minCoord">The coordinates of the top left of the icon area.</param>
+    /// <param name="maxCoord">The coordinates of the bottom right of the icon area.</param>
+    /// <param name="textureTask">The task that results in a texture.</param>
+    /// <returns><c>true</c> if anything has been drawn.</returns>
+    /// <remarks>Exceptions from the task will be treated as if no texture is provided.</remarks>
+    internal static bool DrawIconFrom(Vector2 minCoord, Vector2 maxCoord, Task<IDalamudTextureWrap?>? textureTask) =>
+        textureTask?.IsCompletedSuccessfully is true && DrawIconFrom(minCoord, maxCoord, textureTask.Result);
+
+    /// <summary>Draws an icon from an instance of <see cref="LocalPlugin"/>.</summary>
+    /// <param name="minCoord">The coordinates of the top left of the icon area.</param>
+    /// <param name="maxCoord">The coordinates of the bottom right of the icon area.</param>
+    /// <param name="plugin">The plugin. Dalamud icon will be drawn if <c>null</c> is given.</param>
+    /// <returns><c>true</c> if anything has been drawn.</returns>
+    internal static bool DrawIconFrom(Vector2 minCoord, Vector2 maxCoord, LocalPlugin? plugin)
+    {
+        var dam = Service<DalamudAssetManager>.Get();
+        if (plugin is null)
+            return false;
+
+        if (!Service<PluginImageCache>.Get().TryGetIcon(
+                plugin,
+                plugin.Manifest,
+                plugin.IsThirdParty,
+                out var texture) || texture is null)
+        {
+            texture = plugin switch
+            {
+                { IsDev: true } => dam.GetDalamudTextureWrap(DalamudAsset.DevPluginIcon),
```

Comment:
Would it make sense to just draw the "question mark" icon instead? These are just overlays, might look a little odd.